src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfacePromptForChoice.cs

```diff
@@ -361,35 +327,77 @@ public override int PromptForChoice(string caption, string message, Collection<C
                 }
                 else
                 {
-                    defaultPrompt = StringUtil.Format(ConsoleHostUserInterfaceStrings.DefaultChoicesForMultipleChoices,
+                    defaultPrompt = StringUtil.Format(
+                        ConsoleHostUserInterfaceStrings.DefaultChoicesForMultipleChoices,
                         defaultChoices);
                 }
-            }
 
-            WriteChoiceHelper(defaultPrompt,
-                fg,
-                bg,
-                ref lineLen,
-                lineLenMax);
+                WriteToConsole(
+                    PSStyle.Colorize(defaultPrompt, PSStyle.Instance.Prompt.Help) + " ",
+                    transcribeResult: true);
+            }
         }
 
-        private void WriteChoiceHelper(string text, ConsoleColor fg, ConsoleColor bg, ref int lineLen, int lineLenMax)
+        /// <summary>
+        /// Joins lines with separator, breaking into new line if longer than window width.
+        /// </summary>
+        /// <param name="separator">
+        /// Separator used between strings fitting on the same line.
+        /// </param>
+        /// <param name="values">
+        /// Collection of strings, never broken unless cannot fit on one line by itself.
+        /// </param>
+        /// <returns>
+        /// Joined string.
+        /// </returns>
+        private string JoinWrappedToCurrentWindowWidth(string separator, IEnumerable<string> values)
         {
-            int textLen = RawUI.LengthInBufferCells(text);
-            bool trimEnd = false;
+            // Avoid recalculating every time, as skipping over escapes is nontrivial
+            int lineLength = 0;
+            int maxLength = RawUI.WindowSize.Width;
 
-            if (lineLen + textLen > lineLenMax)
+            var result = new StringBuilder();
+            foreach (var value in values)
             {
-                WriteLineToConsole();
-                trimEnd = true;
-                lineLen = textLen;
+                int valueLength = RawUI.LengthInBufferCells(value);
```

Comment:
Users can call `$Host.Ui.PromptForChoice()` in their scripts and embed ANSI in the choice values, so the length calculation here would not be correct.  You'll probably have to use https://github.com/PowerShell/PowerShell/blob/3d4e294262d2f2d7472ebd5d849b1b68fd6ef524/src/System.Management.Automation/FormatAndOutput/common/StringDecorated.cs#L50 to get the printable length.  I tried a simple example with the current code:

```powershell
$choices = ($psstyle.foreground.green+1,$psstyle.foreground.red+2,$psstyle.foreground.magenta+3)
$Host.UI.PromptForChoice("hello","world",$choices,-1)
```

And it's currently broken for any default choice index >= 0 and doesn't render correctly.  I don't think you need to fix that within this PR, so perhaps for now if we detect choice values have ANSI, we throw an error saying it's not supported (you can use the StringDecorated class to detect that as well)

Reply:
1. I thought `LengthInBufferCells` is exactly for getting printable length, if not, what is it for?
1. Your example is not valid, as PowerShell's stupid operator precendence of `,` above `+` makes `$choices` a single string. What you probably meant is `$choices = @(($psstyle.foreground.green+1),($psstyle.foreground.red+2),($psstyle.foreground.magenta+3))`. This shows that the only bug is existing lack of reset at the end of default help:
    before change:
![image](https://user-images.githubusercontent.com/26599495/191340445-24425f6b-8aa0-41cf-b3f8-4ef8d1893b44.png)
    after change:
![image](https://user-images.githubusercontent.com/26599495/191338396-6d57b36a-f281-42ea-8028-e1c0dc0c64f1.png)
    fixed:
![image](https://user-images.githubusercontent.com/26599495/191340850-76138be5-1607-4bbc-b8b7-3b529d8d137f.png)

Reply:
> I thought LengthInBufferCells is exactly for getting printable length, if not, what is it for?

Historically for wide chars.

Reply:
Looks like ConsoleHost implementation of `LengthInBufferCells` has some basic checking for escape sequences, but other hosts may not.  I think it would be better to use `StringDecorated` to get the plaintext version to calculate buffercells.

Reply:
Changed `LengthInBufferCells(value)` to `new StringDecorated(value).ContentLength`